### DISCUSSION TITLE
Update sketchware blocks system

### DISCUSSION THREAD

**Discussion by halloweeks** [Sketchware-Pro/Sketchware-Pro#747] [Ideas]
Update sketchware blocks system like kodular
Kodular block system is very easy to understand 

![images (3)](https://user-images.githubusercontent.com/96823158/216949542-cd5d0d7c-9310-4f2e-9351-218e71e836ac.jpeg)

![images (1)](https://user-images.githubusercontent.com/96823158/216948898-2a5e6696-4a0d-4d92-a53d-7df1251c1dcb.png)

**Comment by DevVigilante** [comment]
This can be possible if there is option to create custom block as regular block,if-block.
Example
![20230206_192446](https://user-images.githubusercontent.com/96680248/216989803-14f3cf85-def1-4359-afc0-47f4d8814e9f.jpg)
@hasrat-ali, @JavkhlanK please think about it also🙂

**Comment by hasrat-ali** [comment]
I would absolutely not want that the block system should be tampered with. Because still we do not fully know how it works.

**Comment by ilyassesalama** [comment]
This is not recommended at all since the majority of Sketchware users are used to the original design (system), so modders can either improve the current system or not touch it at all.
There are more important things to work on currently such as deobfuscating the remaining classes as they will help to make the development of Sketchware Pro faster and easier for everyone.

**Reply by hasrat-ali** [reply]
What's remains for de-obfuscating?

Did you mean decompiling?

**Reply by hasrat-ali** [reply]
Do you mean all classes in  `com.besome.sketch.*` package?
